app/javascript/lib/search-utils.js

```diff
@@ -0,0 +1,48 @@
+import stringSimilarity from 'string-similarity'
+
+/**
+ * Get list of autocomplete suggestions, based on input text
+ *
+ * Returns top matches: exact prefix matches, then similar matches
+ *
+ * @param {String} inputString String typed by user into text input
+ * @param {Array<String>} targets List of strings to match against
+ * @param {Integer} numSuggestions Number of suggestions to show
+ */
+export function getAutocompleteSuggestions(inputText, targets, numSuggestions=8) {
+  // Autocomplete when user starts typing
+  if (!targets || targets.length === 0 || !inputText) {
```

Comment:
FYI that this is a case where you could just do 
`if (!targets?.length || !inputText) { `

Reply:
Neat, done in 1261991.